File: assets/samples/dns.lookup/password/nodejs/dns_lookup_pass.js
var http = require('http');

var domain = 'example.com';
var password = 'your whois api password';
var username = 'your whois api username';

var url = 'http://www.whoisxmlapi.com/whoisserver/DNSService?type=_all'
        + '&domainName=' + domain + '&username=' + username + '&password=' + password;

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) { str += chunk; });
    response.on('end', function() { console.log(str); });
}).end();

File: assets/samples/dns.lookup/password/ruby/dns_lookup_pass.rb
require 'open-uri'
require 'json'

username = 'your whois api username'
password = 'your whois api password'
domain =  'twitter.com'
type = '_all'
format = 'json'
url = 'https://whoisxmlapi.com/whoisserver/DNSService?'
url += 'type=' + type + '&username=' + username
url += '&password=' + password + '&outputFormat=' + format
url += '&domainName=' + domain

def print_response(response)
  response_hash = JSON.parse(response)
  puts JSON.pretty_generate(response_hash)
end

response = open(url).read
print_response(response)